src/ucp/core/ucp_worker.c

```diff
@@ -1297,6 +1371,15 @@ ucs_status_t ucp_worker_iface_open(ucp_worker_h worker, ucp_rsc_index_t tl_id,
         goto err_close_iface;
     }
 
+    /* save attributes adjusted by distance from mem_type MD */
+    memcpy(&wiface->adjusted_attr, &wiface->attr, sizeof(wiface->attr));
+
+    status = ucp_worker_get_sys_device_distance(context, wiface->rsc_index,
+                                                &distance);
+    if (status == UCS_OK) {
+        wiface->adjusted_attr.latency.c += distance.latency;
```

Comment:
what about bw = ucs_min(bw, distance.bw)
(for both dedicated and shared)

Reply:
Adding bandwidth increment based on distance reported.